--- prboom-2.3.0/src/d_main.c.orig	2003-08-31 14:40:57.000000000 +0200
+++ prboom-2.3.0/src/d_main.c	2003-10-08 23:08:05.961283192 +0200
@@ -956,7 +956,11 @@
 /* proff 2001/7/2 - On Dreamcast the free crashes. There is a bug somewhere. */
     free(iwad);
 #endif
-    D_AddFile("prboom.wad",source_pwad);
+    iwad = I_FindFile("prboom",".wad");
+    if(iwad && *iwad)
+      D_AddFile(iwad,source_pwad);
+    else
+      I_Error("prboom.wad not found\n");
   }
   else
     I_Error("IdentifyVersion: IWAD not found\n");
